plugins/module_utils/network/sonic/config/sflow/sflow.py

```diff
@@ -0,0 +1,522 @@
+            if "sampling_rate" in interface:
+                interface_config_request["sampling-rate"] = interface["sampling_rate"]
+            if len(interface_config_request) == 0:
+                # listed interface doesn't actually have any configured settings, but name is hanging around
+                continue
+            interface_config_request["name"] = interface["name"]
+            interface_list.append({"name": interface["name"],
+                                   "config": interface_config_request})
+        return interface_list
+
+    def get_deleted_requests(self, to_delete, have):
+        requests = []
+        if "enabled" in to_delete:
+            requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/config/enabled", "method": "PUT",
+                             "data": {"openconfig-sampling-sflow:enabled": False}})
+        if "polling_interval" in to_delete:
+            requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/config/polling-interval", "method": "DELETE"})
+        if "agent" in to_delete:
+            requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/config/agent", "method": "DELETE"})
+        if "sampling_rate" in to_delete:
+            requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/config/sampling-rate", "method": "DELETE"})
+        if "collectors" in to_delete:
+            have_collectors_dict = {(collector["address"], collector["network_instance"], collector["port"]): collector for collector in have["collectors"]}
+            to_delete_collectors_dict = {(collector["address"], collector["network_instance"], collector["port"]): collector
+                                         for collector in to_delete["collectors"]}
+            if have_collectors_dict.keys() == to_delete_collectors_dict.keys():
+                requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/collectors", "method": "DELETE"})
+            else:
+                for collector in to_delete["collectors"]:
+                    requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/collectors/collector=" +
+                                    collector["address"] + "," + str(collector["port"]) + "," +
+                                    collector["network_instance"], "method": "DELETE"})
+        if "interfaces" in to_delete:
+            for interface in to_delete["interfaces"]:
+                requests.append({"path": "data/openconfig-sampling-sflow:sampling/sflow/interfaces/interface=" + interface["name"], "method": "DELETE"})
+        return requests
+
+    def fill_defaults(self, config):
+        '''modifies the given original config object to add sflow default values that are missing. returns the config for chaining purposes'''
+        if "enabled" not in config:
+            config["enabled"] = False
+        return config
+
+    def find_substitution_deletes(self, remove_diff, introduced_diff):
```

Comment:
The name of this function would match what it does more closely if it were changed to something that highlights the fact that it removes items from the input remove_diff if they are deletes corresponding to a substitution. (The current name sounds like just the opposite.)

e.g. "find_non_substitution_deletes" or "remove_substitution_deletes".